src/librustc/ty/print/pretty.rs

```diff
@@ -892,17 +899,25 @@ pub trait PrettyPrinter<'tcx>:
                     }
                 }
             }
-            (ty::ConstKind::Infer(..), _) => print_underscore!(),
-            (ty::ConstKind::Param(ParamConst { name, .. }), _) => p!(write("{}", name)),
-            (ty::ConstKind::Value(value), _) => {
+            ty::ConstKind::Infer(..) => print_underscore!(),
+            ty::ConstKind::Param(ParamConst { name, .. }) => p!(write("{}", name)),
+            ty::ConstKind::Value(value) => {
                 return self.pretty_print_const_value(value, ct.ty, print_ty);
             }
 
-            _ => {
+            ty::ConstKind::Bound(..) | ty::ConstKind::Placeholder(_) => {
```

Comment:
What does `Ty` printing do for these two? We should do something similar (they're like unnamed parameters IIRC).

Reply:
I went through all mentions of `ConstKind::Bound` in the source, there are no others that do printing. This is the single location where they are printed.

Reply:
No, `TyKind::{Bound,Placeholder}` - these are paralleled between types and consts just like inference variables and generic params.